src/tests/unit/auto/auto_ctput_test.cpp

```diff
@@ -0,0 +1,246 @@
+            return plugin->MultiDeviceInferencePlugin::GetDeviceList(config);
+        });
+        ON_CALL(*plugin, SelectDevice)
+            .WillByDefault([this](const std::vector<DeviceInformation>& metaDevices,
+                                  const std::string& netPrecision,
+                                  unsigned int Priority) {
+                return plugin->MultiDeviceInferencePlugin::SelectDevice(metaDevices, netPrecision, Priority);
+            });
+        std::vector<std::string> cpuCability{"FP32", "FP16", "INT8", "BIN"};
+        std::vector<std::string> gpuCability{"FP32", "FP16", "BATCHED_BLOB", "BIN", "INT8"};
+        ON_CALL(*core, GetMetric(StrEq(CommonTestUtils::DEVICE_CPU), StrEq(METRIC_KEY(OPTIMIZATION_CAPABILITIES)), _))
+            .WillByDefault(Return(cpuCability));
+        ON_CALL(*core, GetMetric(StrEq(CommonTestUtils::DEVICE_GPU), StrEq(METRIC_KEY(OPTIMIZATION_CAPABILITIES)), _))
+            .WillByDefault(Return(gpuCability));
+
+        ON_CALL(*core,
+                LoadNetwork(::testing::Matcher<const InferenceEngine::CNNNetwork&>(_),
+                            ::testing::Matcher<const std::string&>(StrEq("CPU")),
+                            ::testing::Matcher<const std::map<std::string, std::string>&>(_)))
+            .WillByDefault(Return(cpuMockExeNetwork));
+
+        ON_CALL(*core,
+                LoadNetwork(::testing::Matcher<const InferenceEngine::CNNNetwork&>(_),
+                            ::testing::Matcher<const std::string&>(StrEq("GPU")),
+                            ::testing::Matcher<const std::map<std::string, std::string>&>(_)))
+            .WillByDefault(Return(gpuMockExeNetwork));
+
+        std::shared_ptr<ngraph::Function> simpleNetwork = ngraph::builder::subgraph::makeSingleConv();
+        ASSERT_NO_THROW(simpleCnnNetwork = InferenceEngine::CNNNetwork(simpleNetwork));
+    }
+};
+
+TEST_P(LoadNetworkWithCTPUTMockTest, CTPUTSingleDevLogicTest) {
+    std::vector<std::string> targetDevices;
+    Config config;
+    std::tie(targetDevices) = this->GetParam();
+
+    plugin->SetName("AUTO");
+    config.insert({{CONFIG_KEY(PERFORMANCE_HINT), InferenceEngine::PluginConfigParams::CUMULATIVE_THROUGHPUT}});
+
+    if (targetDevices.size() == 1) {
+        std::string targetDevice = targetDevices[0];
+        config.insert({InferenceEngine::MultiDeviceConfigParams::KEY_MULTI_DEVICE_PRIORITIES, targetDevices[0]});
+        // Call single device logic and performance hint is THROUGHPUT
+        EXPECT_CALL(*core,
+                    LoadNetwork(::testing::Matcher<const InferenceEngine::CNNNetwork&>(_),
+                                ::testing::Matcher<const std::string&>(targetDevice),
+                                ::testing::Matcher<const std::map<std::string, std::string>&>(
+                                    ComparePerfHint(InferenceEngine::PluginConfigParams::THROUGHPUT))))
+            .Times(1);
+        // if target device only has GPU, no CPU helper to be called
```

Comment:
also expect_call to core->loadnetwork("MULTI") is 0

Reply:
updated